lib/launchers/process.js

```diff
@@ -101,6 +102,9 @@ function ProcessLauncher (spawn, tempDir, timer, processKillTimeout) {
   }
 
   this._onProcessExit = function (code, signal, errorOutput) {
+    if (!self._process) {
```

Comment:
This prevents executing this method twice in case both error and exit event are emitted to comply with https://nodejs.org/api/child_process.html#child_process_event_error "The 'exit' event may or may not fire after an error has occurred. When listening to both the 'exit' and 'error' events, guard against accidentally invoking handler functions multiple times."

Reply:
Please move this information into a comment in the code, eg

// Both `exit` and `error` events trigger _onProcessExit(), but we only need one cleanup.

Reply:
Thanks, comment added.